photutils/detection/findstars.py

```diff
@@ -7,18 +7,20 @@
 import warnings
 import math
 import numpy as np
+import abc
 from astropy.table import Column, Table
 from astropy.utils.exceptions import AstropyUserWarning
 from astropy.stats import gaussian_fwhm_to_sigma
 from .core import _convolve_data, find_peaks
 
 
-__all__ = ['daofind', 'irafstarfind']
+class StarFinder(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def __call__(self, data):
```

Comment:
It's a bit of an open question given that lengthy discussion from thursday... but I'm inclined towards having a method called ``find_stars`` that's the "real" method that does the star-finding, and then have the `__call__` just be:

```
def __call__(self, *args, **kwargs):
    return self.find_stars(*args, **kwargs)
```

Then we tell users the normal way to override this class is to override ``find_stars``, not ``__call__``